src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -914,6 +914,129 @@ def get_bumps(self) -> Optional[BumpsDefinition]:
             pitch=self.get_setting("vlsi.inputs.bumps.pitch"),
             cell=self.get_setting("vlsi.inputs.bumps.cell"), assignments=assignments)
 
+    def generate_floorplan_viz(self) -> None:
+        viz = self.get_setting("pcb.generic.viz")
+        if viz in ["all", "bumps"]:
+            fsvg = open(os.path.join(self.run_dir, 'floorplan.svg'), 'w')
+            fsvg.write("""<?xml version="1.0"?>
+            <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\n""")
+
+            title_height = 100
+            fp_consts = self.get_placement_constraints()
```

Comment:
Don't you need to check that there is something here?
What happens if someone hasn't specified a floorplan but asks for a viz?

Reply:
Updated to check for empty list and throwing a logging error. However, if no floorplans are specified, other things will break in a par run anyway (like bumps).

Reply:
You mean if you specify bumps and no floorplan? Or if you specify no bumps and no floorplan? Can you submit an issue on that